Dockerfile

```diff
@@ -1,117 +1,155 @@
-	make -j$(nproc) > /dev/null && \
-	make install
-
-ENV PATH="${PATH}:/opt/ruby/bin:/opt/node/bin"
-
+RUN apt-get update && apt-get --no-install-recommends -y install \
+    build-essential \
+    bison \
+    libgdbm-dev \
+    libncurses5-dev \
+    libreadline-dev \
+    libffi-dev \
+    libssl-dev \
+    libyaml-dev \
+    zlib1g-dev 
+RUN cd ~ && \
+    wget https://cache.ruby-lang.org/pub/ruby/${RUBY_VER%.*}/ruby-${RUBY_VER}.tar.gz && \
+    tar xf ruby-${RUBY_VER}.tar.gz && \
+    cd ruby-${RUBY_VER} && \
+    ./configure --prefix=/opt/ruby \
+    --with-jemalloc \
+    --with-shared \
+    --disable-install-doc && \
+    ln -s /opt/jemalloc/lib/* /usr/lib/ && \
+    make -j$(nproc) > /dev/null && \
+    make install
+
+# Update PATH
+ENV PATH="/opt/ruby/bin:/opt/node/bin:${PATH}"
+
+# Install mastodon package managers
 RUN npm install -g yarn && \
-	gem install bundler && \
-	apt update && \
-	apt -y install git libicu-dev libidn11-dev \
-	libpq-dev libprotobuf-dev protobuf-compiler
+    gem install bundler
+
+# Install mastodon OS Package
+RUN apt-get update && apt-get --no-install-recommends -y install \
+    git \
+    libicu-dev \
+    libidn11-dev \
+    libpq-dev \
+    libprotobuf-dev \
+    protobuf-compiler
 
 COPY Gemfile* package.json yarn.lock /opt/mastodon/
 
 RUN cd /opt/mastodon && \
-  bundle config set deployment 'true' && \
-  bundle config set without 'development test' && \
-	bundle install -j$(nproc) && \
-	yarn install --pure-lockfile
+    bundle config set deployment 'true' && \
+    bundle config set without 'development test' && \
+    bundle install -j$(nproc) && \
+    yarn install --pure-lockfile
 
+##-------------------------------------------------
 FROM ubuntu:18.04
 
+# Use bash for the SHELL
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
+RUN echo "Etc/UTC" > /etc/localtime && \
+    apt-get update && apt-get --no-install-recommends -y install \
+    apt-utils \
+    wget \
+    whois \
+    ca-certificates
+
 # Copy over all the langs needed for runtime
 COPY --from=build-dep /opt/node /opt/node
 COPY --from=build-dep /opt/ruby /opt/ruby
 COPY --from=build-dep /opt/jemalloc /opt/jemalloc
+RUN ln -s /opt/jemalloc/lib/* /usr/lib/
 
-# Add more PATHs to the PATH
-ENV PATH="${PATH}:/opt/ruby/bin:/opt/node/bin:/opt/mastodon/bin"
+# Update PATH
+ENV PATH="/opt/ruby/bin:/opt/node/bin:/opt/mastodon/bin:${PATH}"
 
 # Create the mastodon user
 ARG UID=991
 ARG GID=991
-RUN apt update && \
-	echo "Etc/UTC" > /etc/localtime && \
-	ln -s /opt/jemalloc/lib/* /usr/lib/ && \
-	apt install -y whois wget && \
-	addgroup --gid $GID mastodon && \
-	useradd -m -u $UID -g $GID -d /opt/mastodon mastodon && \
-	echo "mastodon:`head /dev/urandom | tr -dc A-Za-z0-9 | head -c 24 | mkpasswd -s -m sha-256`" | chpasswd
+RUN addgroup --gid ${GID} mastodon && \
+    useradd -m -u ${UID} -g ${GID} -d /opt/mastodon mastodon && \
+    echo "mastodon:$(head /dev/urandom | tr -dc A-Za-z0-9 | head -c 24 | mkpasswd -s -m sha-256)" | chpasswd
```

Comment:
we could just add the account with no password.

Reply:
I'm really not sure why that exists... but again. this code review is just for readability.

If it ain't broke.. leave it for another PR.